lib/spack/llnl/util/filesystem.py

```diff
@@ -1180,6 +1187,69 @@ def remove_if_dead_link(path):
         os.unlink(path)
 
 
+class GenericVisitor(object):
+    def __init__(self, callable):
+        pass
+
+    def before_visit_dir(self, root, rel_path, depth):
+        pass
```

Comment:
It looks like this is an interface, so:

* The name `Generic` is misleading: `VisitorInterface` would be more useful
* These functions should raise `NotImplementedError`
* The interface should have a docstring for the class and for each function explaining the intent

Reply:
The intent was to create a stub class rather than a true interface as per [this comment](https://github.com/spack/spack/pull/29714#discussion_r838676576), however if you feel we should be implementing a true interface class here that's fine with me.